plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -342,26 +321,31 @@ func getServiceStatus(lteMetric lteDeviceState) string {
 
 func getSignalValue(lteMetric lteDeviceState) map[string]string {
 	signalStrengths := make(map[string]string)
-	values := reflect.ValueOf(lteMetric.LTE)
-	typesOf := values.Type()
-	for i := 0; i < values.NumField(); i++ {
-		if strings.Contains(typesOf.Field(i).Name, "SignalStrength") && values.Field(i).Interface() != "" {
-			signalKey := getSignalKeyName(typesOf.Field(i).Name)
-			signalStrengths[signalKey] = strings.Split(values.Field(i).Interface().(string), " ")[0]
-		}
+	if lteMetric.LTE.RSRPSignalStrength != "" {
+		signalKey := getSignalKeyName("RSRPSignalStrength")
+		signalStrengths[signalKey] = strings.Split(lteMetric.LTE.RSRPSignalStrength, " ")[0]
+	}
+	if lteMetric.LTE.RSRQSignalStrength != "" {
+		signalKey := getSignalKeyName("RSRQSignalStrength")
+		signalStrengths[signalKey] = strings.Split(lteMetric.LTE.RSRQSignalStrength, " ")[0]
```

Comment:
This logic is repeated a lot. It could easily be:
```go
signalStrengths["rsrp"] = getSignalValue(lteMetric.LTE.RSRPSignalStrength)

def getSignalValue(value string) string {
    return strings.Split(value, " ")[0]
}
```
Also, as you'll notice there, you don't have a reason to write out some long string (`RSRPSignalStrength`) just to pass it to a function that does a known transform on it. Just write down `rsrp` to start with...